src/sinol_make/commands/run/__init__.py

```diff
@@ -183,8 +183,8 @@ def compile_solutions(self, solutions):
         print("Compiling %d solutions..." % len(solutions))
         with mp.Pool(self.cpus) as pool:
             compilation_results = pool.map(self.compile, solutions)
-        if not all(compilation_results):
-            util.exit_with_error("\nCompilation failed.")
+        if len(list(filter(lambda x: x, compilation_results))) == 0:
+            util.exit_with_error("\nNo solutions were compiled.")
```

Comment:
I'm not sure whether I understand this code. Sinol-make should exit with an error when at least one solution failed to compiled, not when all solutions failed to compile. That should be done at the end, after printing the table etc.

Reply:
The reasoning is that we want to me kind as much as possible and print the table even when something is wrong with the solutions, but nevertheless we should return with a non-zero exit code so that CI etc. doesn't pass.

Reply:
When some solutions fail to compile the code will exit with error at the end ([here](https://github.com/sio2project/sinol-make/blob/9b9ad252afc407846bcf4a06d294715461e329c6/src/sinol_make/commands/run/__init__.py#L722)), but this two lines exit when every solution had compilation errors

Reply:
So when no solution compiles successfully, the executions won't start

Reply:
Is there a reason for handling this edge-case? E.g. does the output look bad for some reason when no solution compiled? As long as it doesn't help the user, we should try to make the code shorted so it's easier to maintain.

Reply:
Maybe this line would be more readable: `if not any(compilation_results):`

Reply:
The output looks like this without the if:
```
Compilation of file wcf.cpp was unsuccessful.
/mnt/c/Users/maslo/oi/sinol-make/tests/packages/wcf/prog/wcf.cpp: In function ‘int main()’:
/mnt/c/Users/maslo/oi/sinol-make/tests/packages/wcf/prog/wcf.cpp:7:9: error: unused variable ‘c’ [-Werror=unused-variable]
    7 |     int c; // Unused variable
      |         ^
cc1plus: all warnings being treated as errors

Performing 0 executions...
Expected scores from config:
sinol_expected_scores: {}


Expected scores based on results:
sinol_expected_scores: {}

Expected scores are correct!
Compilation failed for 1 solution(s).
```

Reply:
Doesn't look necessarily bad, it can stay like this. That would shorten our code and reduce chances of a bug.

If we want to make some if's so that the output is nicer, I would say that we can shorten the "expected scores output". There is no reason to print both tables when they are equal. But that is better left for another PR.